repose-aggregator/components/filters/rackspace-identity-basic-auth/src/main/scala/org/openrepose/filters/rackspaceidentitybasicauth/RackspaceIdentityBasicAuthFilter.scala

```diff
@@ -100,19 +98,34 @@ class RackspaceIdentityBasicAuthFilter @Inject()(configurationService: Configura
   }
 
   override def doFilter(servletRequest: ServletRequest, servletResponse: ServletResponse, filterChain: FilterChain) {
-    new FilterLogicHandlerDelegate(servletRequest, servletResponse, filterChain).doFilter(this)
+    val wrappedHttpRequest = new HttpServletRequestWrapper(servletRequest.asInstanceOf[HttpServletRequest])
+    val httpResponse = servletResponse.asInstanceOf[HttpServletResponse]
+
+    val filterAction = handleRequest(wrappedHttpRequest, httpResponse)
+    filterAction match {
```

Comment:
That's true. I added all of them for the sake of completeness, and to maintain the behavior using the FilterLogicHandlerDelegate would have provided. Well, except for NOT_SET, which is no longer applicable. I'm actually not sure if the Scala compiler requires all values of an enum to be represented in a match as it does with sealed traits.